### PR TITLE
Initial commit - database creation working as expected

### PR DESCRIPTION
**Pull request by isuftin**
This image is based on the latest official postgres Docker image. Creates the database based on dump files. Will download the latest dump files if they don't exist.

Future work would be to test if user wishes to create database fresh (no dump files). Also, not validating dump files from server.

Included docker-compose file to kick off initial run after image is built. Used your DBDefs.pm file from the server image source.

### CODE REVIEW

File: musicbrainz-postgres/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+COPY /scripts/init-database.sh /docker-entrypoint-initdb.d/
+
+COPY /scripts/initscript.sh /tmp/initscript.sh
+
+COPY /scripts/DBDefs.pm /tmp/DBDefs.pm
+
+COPY /scripts/cpan-deps.sh /tmp/cpan-deps.sh
+
+VOLUME ["/dump"]
```
Comment (isuftin) [inline musicbrainz-postgres/Dockerfile]: dump files will be put into this directory.

File: musicbrainz-postgres/scripts/DBDefs.pm [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+# The Database
+################################################################################
+
+# Configuring databases here is required; there are no defaults.
+MusicBrainz::Server::DatabaseConnectionFactory->register_databases(
+    # How to connect when we need read-write access to the database
+    READWRITE => {
+        database    => "musicbrainz",
+        schema      => "musicbrainz",
+        username    => "musicbrainz",
```
Comment (isuftin) [inline musicbrainz-postgres/scripts/DBDefs.pm:56]: In the next iteration, should probably do sed replace on username/pass based on incoming ENV variables

**Comment by rickatnight11** [comment]
Apologies for the delay, @isuftin.  I had a `musicbrainz-postgres` build all set up and never got around to pushing it until now.  I'm not yet using the official postgres container, but want to take a look at what I just merged to master and tweak accordingly?

**Comment by isuftin** [comment]
@rickatnight11 I will do so the first chance I get. I think I'v also included some image build-time stuff that doesn't need to be in there so I'd like to refine it a bit. 

Thanks for looking into it!

**Comment by isuftin** [comment]
I'm going to re-work my image a bit and re-create the PR. I'd like to get the database image working with the official postgres container. Hoping to have something to look at soon.

**Pull request closed without merging**